editionupdate

pheobeayo opened the pull request:
<!-- Reference any GitHub issues resolved by this PR -->

Closes #

## Introduced changes

<!-- A brief description of the changes -->

-migrate scarb package to edition 2024_07

## Checklist

<!-- Make sure all of these are complete -->

- [https://github.com/foundry-rs/starknet-foundry/issues/2651 ] Linked relevant issue 
- [ ] Updated relevant documentation
- [ ] Added relevant tests
- [ ] Performed self-review of the code
- [ ] Added changes to `CHANGELOG.md`

franciszekjob commented:
Hi @pheobeayo, please check out my last comment in #2651 :)

pheobeayo commented:
@franciszekjob Okay

cptartur reviewed (changes requested):
Doesn't resolve the linked issue

pheobeayo commented on snforge_std/Scarb.lock:
Okay

franciszekjob commented:
@pheobeayo edition need to be changed in other packages as well, please verify it. If you have any questions, feel free to ask :).

pheobeayo commented:
@franciszekjob Kindly check my last commits,  please. I made changes in some other packages. Kindly point out others I didn't pick out if all the changes haven't been made

franciszekjob commented:
@pheobeayo there are many more packages, e.g. under `crates/forge/tests/data`.

franciszekjob commented:
Hi @pheobeayo do you need any help?

pheobeayo commented:
@franciszekjob I need help fixing the errors on the CI, I have been trying to resolve it since but I haven't gotten any headway, any clue?

franciszekjob commented:
> @franciszekjob I need help fixing the errors on the CI, I have been trying to resolve it since but I haven't gotten any headway, any clue?

Bumping edition results in occurring errors (which are visible on CI, [example](https://github.com/foundry-rs/starknet-foundry/actions/runs/11933249431/job/33319887242)). Alternatively, you can open any package in which you bumped `edition` and run `scarb build` there - you will see these errors. In scope of this task, we should:
1. bump `edition` (this is mostly done)
2. resolve the errors

Hope this helps you 😄

franciszekjob commented:
Hi @pheobeayo do you have any updates?

pheobeayo commented:
> Hi @pheobeayo do you have any updates?

@franciszekjob Yes, I do have updates, I have fixed most of the bugs except for this...

![image](https://github.com/user-attachments/assets/ece14faf-4738-4f1c-8816-239738c58782)
any clue on what to do here, everything I implemented didn't resolve this issue

pheobeayo commented:
> @pheobeayo could you show the code of contract?

@franciszekjob I have been able to solve that, 

I am got stuck here which is the final bug

pheobeayo commented:
@franciszekjob I Have made the push

pheobeayo commented:
@franciszekjob I have updated the PR and fixed other errors

cptartur reviewed (changes requested):
Most of the tests are still failing. See "Some checks were not successful" section.

franciszekjob reviewed (changes requested):
@pheobeayo CI is still failing
https://github.com/foundry-rs/starknet-foundry/pull/2681/checks

pheobeayo commented:
I was just about to push the latest changes I made to it

pheobeayo commented:
@ should I create a new PR and make the push @franciszekjob, I have been working on the test failure
